src/error-handling/try-operator.md

```diff
@@ -20,26 +20,31 @@ We can use this to simplify our error handing code:
 
 ```rust,editable
 use std::fs;
-use std::io::{self, Read};
+use std::io::{self, Error, ErrorKind, Read};
 
-fn read_username(path: &str) -> Result<String, io::Error> {
-    let username_file_result = fs::File::open(path);
-    let mut username_file = match username_file_result {
+// Reads the email address from a file and returns the username and hostname.
+fn read_email_address(path: &str) -> Result<(String, String), io::Error> {
+    let email_file_result = fs::File::open(path);
+    let mut email_file = match email_file_result {
         Ok(file) => file,
         Err(err) => return Err(err),
     };
 
-    let mut username = String::new();
-    match username_file.read_to_string(&mut username) {
-        Ok(_) => Ok(username),
-        Err(err) => Err(err),
+    let mut email = String::new();
+    if let Err(err) = email_file.read_to_string(&mut email) {
+        return Err(err);
+    }
+    if let Some((username, hostname)) = email.split_once('@') {
```

Comment:
The extra complexity of splitting and returning a `(String, String)` tuple seems unnecessary to me.

The example is supposed to focus laser-sharp on transforming `match foo { Ok(foo) => foo, Err(err) => return Err(err) }` to `foo?`. There should be very very little business logic apart from that.

Reply:
I need to illustrate how to transform `Option` to `Result`. I thought `split` was a good example in that case because missing the `@` in the string would mean invalid email address.  Is it fine to drop the `tuple` and only return the username?

Reply:
Oh... :sweat_smile:  See, I had not even noticed that this was the point of the change.

We've started putting larger and larger examples in the speaker notes over the last few months. Basically, when teaching the class, it's very poor style when we have to scroll up and down to explain an example. This is what the [aspect ratio helper](https://github.com/google/comprehensive-rust/blob/main/STYLE.md#vertical-space) is for and I suggest you check it out.

This is all to say that we should not add more vertical space to the slide. We should instead mention these point (important and great points!) in the speaker notes.

> Is it fine to drop the `tuple` and only return the username?

If this can be done without adding more vertical space, then sure! Alternatively, we could add a new page for to explicitly mention the most common conversion methods. That would help students by showing the explicit nature of Rust.